### PR TITLE
4.6.0 Release

### PR DESCRIPTION
**Pull request by polqf**
### ⚠️ Important
- Dependencies are no longer exposed (this includes Nuke, SwiftyGif and Starscream). If you were using those dependencies we were exposing, you would need to import them manually. This is due to our newest addition supporting Module Stable XCFrameworks, see more below in the "Added" section.

### 🔄 Changed
- Change `ChatMessageLayoutOptions` to a `Set` instead of an `OptionSet` for a more flexible and safer customization [#1651](https://github.com/GetStream/stream-chat-swift/issues/1651)
- There is a new `ChatMessageListDateSeparatorView` component that should be used instead of the `ChatMessageListScrollOverlayView` if the goal is customize the styling of the date separator. Read [here](https://getstream.io/chat/docs/sdk/ios/uikit/components/message/#date-separators) for more details.

### 🐞 Fixed
- Fix `stopTyping` can be called on `TypingEventSender` after calling `startTyping` [#1649](https://github.com/GetStream/stream-chat-swift/issues/1649)
- Reactions no longer cover the text in message bubble [#1666](https://github.com/GetStream/stream-chat-swift/pull/1666)
- Fix `error` type messages rendered as user's messages and interactive [#1672](https://github.com/GetStream/stream-chat-swift/issues/1672)
- Fix `ChannelListController` makes one redundant API call [#1687](https://github.com/GetStream/stream-chat-swift/issues/1687)
- Safely access indexes of collections [#1692](https://github.com/GetStream/stream-chat-swift/pull/1692)

### ✅ Added
- Add support for pre-built XCFrameworks [#1665](https://github.com/GetStream/stream-chat-swift/pull/1665)
- Added `LogConfig.destinationTypes` for ease of adding new destinations to logger [#1681](https://github.com/GetStream/stream-chat-swift/issues/1681)
- Expose container embedding top & bottom containers by `ChatChannelListItemView` [#1670](https://github.com/GetStream/stream-chat-swift/issues/1670)
- Add Static Message List Date Separators [#1686](https://github.com/GetStream/stream-chat-swift/issues/1686) (You can read this [doc](https://getstream.io/chat/docs/sdk/ios/uikit/components/message/#date-separators) to understand how to configure this feature)

### CODE REVIEW

File: StreamChatArtifacts.json [hunk 1]
```diff
@@ -1 +1 @@
-{}
\ No newline at end of file
+{"4.6.0":"https://github.com/GetStream/stream-chat-swift/releases/download/4.6.0/StreamChat-All.zip"}
```
[the remark this answers is not in the document]
Comment (adamrushy) [reply StreamChatArtifacts.json:1]: Love it 😍

**Review by nuno-vieira — APPROVED** [review]
LGTM! 🚀

**Review by adamrushy — APPROVED** [review]
LGTM 🚀

**Review by adamrushy — DISMISSED** [review]
LGTM 🚀
Do we need to add the upgrade of SwiftyGif to the CHANGELOG?

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,24 +3,34 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
...
+# [4.6.0](https://github.com/GetStream/stream-chat-swift/releases/tag/4.6.0)
+_December 15, 2021_
+
 ### ⚠️ Important
 - Dependencies are no longer exposed (this includes Nuke, SwiftyGif and Starscream). If you were using those dependencies we were exposing, you would need to import them manually. This is due to our newest addition supporting Module Stable XCFrameworks, see more below in the "Added" section.
 
 ### 🔄 Changed
 - Change `ChatMessageLayoutOptions` to a `Set` instead of an `OptionSet` for a more flexible and safer customization [#1651](https://github.com/GetStream/stream-chat-swift/issues/1651)
 - There is a new `ChatMessageListDateSeparatorView` component that should be used instead of the `ChatMessageListScrollOverlayView` if the goal is customize the styling of the date separator. Read [here](https://getstream.io/chat/docs/sdk/ios/uikit/components/message/#date-separators) for more details.
+- `UnknownEvent` is now deprecated, use `UnknownChannelEvent` or `UnknownUserEvent` instead. [#1695](https://github.com/GetStream/stream-chat-swift/pull/1695).
```
Comment (bielikb) [inline CHANGELOG.md:17]:
```suggestion
- `UnknownEvent` is now deprecated, use `UnknownChannelEvent` or `UnknownUserEvent` instead. [#1695](https://github.com/GetStream/stream-chat-swift/pull/1695).
- SwiftyGif now points to [v5.4.2](https://github.com/kirualex/SwiftyGif/releases/tag/5.4.2) that contains delegate.
```

Comment (bielikb) [reply CHANGELOG.md:17]: @adamrushy feel free to adjust the wordings

File: CHANGELOG.md [hunk 3]
```diff
@@ -15,6 +15,7 @@ _December 15, 2021_
 - Change `ChatMessageLayoutOptions` to a `Set` instead of an `OptionSet` for a more flexible and safer customization [#1651](https://github.com/GetStream/stream-chat-swift/issues/1651)
 - There is a new `ChatMessageListDateSeparatorView` component that should be used instead of the `ChatMessageListScrollOverlayView` if the goal is customize the styling of the date separator. Read [here](https://getstream.io/chat/docs/sdk/ios/uikit/components/message/#date-separators) for more details.
 - `UnknownEvent` is now deprecated, use `UnknownChannelEvent` or `UnknownUserEvent` instead. [#1695](https://github.com/GetStream/stream-chat-swift/pull/1695).
+- SwiftyGif now points to [v5.4.2](https://github.com/kirualex/SwiftyGif/releases/tag/5.4.2) that contains delegate.
```
Comment (bielikb) [inline CHANGELOG.md:18]:
```suggestion
- SwiftyGif now points to [v5.4.2](https://github.com/kirualex/SwiftyGif/releases/tag/5.4.2) that resolves crash related to leaked delegate reference.
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,24 +3,35 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
...
+_December 15, 2021_
+
 ### ⚠️ Important
 - Dependencies are no longer exposed (this includes Nuke, SwiftyGif and Starscream). If you were using those dependencies we were exposing, you would need to import them manually. This is due to our newest addition supporting Module Stable XCFrameworks, see more below in the "Added" section.
 
 ### 🔄 Changed
 - Change `ChatMessageLayoutOptions` to a `Set` instead of an `OptionSet` for a more flexible and safer customization [#1651](https://github.com/GetStream/stream-chat-swift/issues/1651)
 - There is a new `ChatMessageListDateSeparatorView` component that should be used instead of the `ChatMessageListScrollOverlayView` if the goal is customize the styling of the date separator. Read [here](https://getstream.io/chat/docs/sdk/ios/uikit/components/message/#date-separators) for more details.
+- `UnknownEvent` is now deprecated, use `UnknownChannelEvent` or `UnknownUserEvent` instead. [#1695](https://github.com/GetStream/stream-chat-swift/pull/1695).
+- SwiftyGif now points to [v5.4.2](https://github.com/kirualex/SwiftyGif/releases/tag/5.4.2) that contains delegate.
```
Comment (bielikb) [inline CHANGELOG.md:18]:
```suggestion
- SwiftyGif now points to [v5.4.2](https://github.com/kirualex/SwiftyGif/releases/tag/5.4.2) that resolves crash related to leaked delegate reference.
```

**Review by bielikb — APPROVED** [review]

**Review by adamrushy — APPROVED** [review]

**Review by b-onc — APPROVED** [review]

**Pull request merged**